csrc/device_lower/pass/alias_memory.cpp

```diff
@@ -1893,6 +1900,20 @@ class PromoteReuseSyncModifier : private kir::ExprMutator {
   void dispatch(Expr* expr) final {
     auto position = allocation_info_map_.getScopeMap().getExprPos(expr);
 
+    // Lifetime intervals are closed, so sync intervals are open. If this is the
+    // first expr past a lifetime's last read, then add the corresponding upper
+    // endpoint for the sync interval. Note that we add these before checking
+    // for first writes so that we can detect adjacent intervals properly.
+    auto range = sync_intervals_.equal_range(position - 1);
```

Comment:
Why is it `position - 1`?

Suppose we have a kernel like:

```
1  float x[N];
2  float y[M];
3  for ...
4   x[...] = ...;
5  end for
6  for ...
7     ... = x[...];
8   end for
9  for ...
10    y[...] = ...;
11 end for
```

And suppose `x` is promoted. What's the sync interval? `[8, 10]`?

Reply:
The sync interval in this case would be (8, 9). We recognize the last read position as the one just before the current expr, which is why we do `position-1`.